tests/interpolation/built_in_resolvers/test_dict.py

```diff
@@ -0,0 +1,136 @@
+from typing import Any
+
+from pytest import mark, param, raises
+
+from omegaconf import ListConfig, OmegaConf
+from omegaconf.errors import InterpolationResolutionError
+
+
+@mark.parametrize(
+    ("cfg", "key", "expected"),
+    [
+        param(
+            {"foo": "${oc.dict.keys:{a: 0, b: 1}}"},
+            "foo",
+            OmegaConf.create(["a", "b"]),
+            id="dict",
+        ),
+        param(
+            {"foo": "${oc.dict.keys:${bar}}", "bar": {"a": 0, "b": 1}},
+            "foo",
+            OmegaConf.create(["a", "b"]),
+            id="dictconfig_interpolation",
+        ),
+        param(
+            {"foo": "${oc.dict.keys:bar}", "bar": {"a": 0, "b": 1}},
+            "foo",
+            OmegaConf.create(["a", "b"]),
+            id="dictconfig_select",
+        ),
+        param(
+            {"foo": "${sum:${oc.dict.keys:{1: one, 2: two}}}"},
+            "foo",
+            3,
+            id="nested",
+        ),
+    ],
+)
+def test_dict_keys(restore_resolvers: Any, cfg: Any, key: Any, expected: Any) -> None:
+    OmegaConf.register_new_resolver("sum", lambda x: sum(x))
+
+    cfg = OmegaConf.create(cfg)
+    val = cfg[key]
+    assert val == expected
+    assert type(val) is type(expected)
+
+    if isinstance(val, ListConfig):
+        assert val._parent is cfg
```

Comment:
Could be a bit cleaner to split into a basic test_dict_keys function and one that tests it with nesting into sum.
Beyond making the test simpler, the input data can also become pure lists (as the expected output type is always ListConfig).

Reply:
Done in 018d521